android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt

```diff
@@ -26,442 +26,486 @@ import com.amplitude.api.Plan
-    val eventWithOptionalProperties: EventWithOptionalPropertiesProperties? = null,
-    val group: GroupProperties? = null,
-    val identify: IdentifyProperties? = null
-)
-
-class ContextProperties()
-
-/**
- * Event to test schema validation
- */
-data class EventMaxIntForTestProperties (
-    /**
-     * property to test schema validation
-     */
-    val intMax10: Long
-)
-
-/**
- * Event w no properties description
- */
-class EventNoPropertiesProperties()
-
-/**
- * Event with Object and Object Array
- */
-data class EventObjectTypesProperties (
-    /**
-     * Property Object Type
-     */
-    val requiredObject: Any?,
-
-    /**
-     * Property Object Array Type
-     */
-    val requiredObjectArray: List<Any?>
-)
-
-/**
- * Event w all properties description
- */
-data class EventWithAllPropertiesProperties (
-    /**
-     * Event 2 Property - Optional String    *     * Examples:    * Some string, or another
-     */
-    val optionalString: String? = null,
-
+class Identify private constructor(
+    eventType: String,
+    eventProperties: Map<String, Any?>?,
+    options: EventOptions? = null
+) : Event(eventType, eventProperties, options) {
     /**
-     * Event 2 Property - Array
-     */
-    val requiredArray: List<String>,
-
-    /**
-     * Event 2 Property - Boolean
-     */
-    val requiredBoolean: Boolean,
-
-    /**
-     * Event 2 Property - Enum
-     */
-    val requiredEnum: EventWithAllPropertiesRequiredEnum,
-
-    /**
-     * Event 2 Property - Integer    *     * Examples:    * 5, 4, 3
-     */
-    val requiredInteger: Long,
-
-    /**
-     * Event 2 Property - Number
+     * Identify
+     *
+     * [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/0.0.0/Identify)
+     *
+     * Identify properties.
+     *
+     * @param requiredNumber Description for identify requiredNumber
+     * @param optionalArray Description for identify optionalArray
      */
-    val requiredNumber: Double,
+    constructor(
+        requiredNumber: Double,
+        optionalArray: Array<String>? = null
+    ) : this(
+        "Identify",
+        mapOf(
+            *(if (optionalArray != null) arrayOf("optionalArray" to optionalArray) else arrayOf()),
+            "requiredNumber" to requiredNumber
+        )
+    )
 
-    /**
-     * Event 2 Property - String
-     */
-    val requiredString: String
-)
+    fun options(userId: String? = null, deviceId: String? = null): Identify {
+        return this.options(EventOptions(userId = userId, deviceId = deviceId))
+    }
 
-/**
- * Event 2 Property - Enum
- */
-enum class EventWithAllPropertiesRequiredEnum {
-    Enum1,
-    Enum2
+    fun options(options: EventOptions): Identify {
+        return Identify(this.eventType, this.eventProperties?.toMap(), options)
+    }
 }
 
-/**
- * Description for event with Array Types
- */
-data class EventWithArrayTypesProperties (
+class EventMaxIntForTest private constructor(
```

Comment:
Even though it's private having the `eventType` settable in the constructor seems strange. Can we do something like the following instead?


```
class EventMaxIntForTest private constructor(
    eventProperties: Map<String, Any?>?,
    options: EventOptions? = null
) : Event(eventType = "EventMaxIntForTest", eventProperties, options) {
 ...
 constructor(
    intMax10: Int
 ) : this(mapOf(
    "intMax10" to intMax10
 ))
}
```

Reply:
Fixed